### PR TITLE
build janus 0.x

### PR DESCRIPTION
**Pull request by Zoey2936**
Address https://github.com/nextcloud/all-in-one/discussions/2881

Fix https://github.com/nextcloud/all-in-one/issues/2996

```
Compiler:                  gcc
libsrtp version:           2.x
SSL/crypto library:        OpenSSL
DTLS set-timeout:          not available
Mutex implementation:      GMutex (native futex on Linux)
DataChannels support:      no
Recordings post-processor: no
TURN REST API client:      no
Doxygen documentation:     no
Transports:
    REST (HTTP/HTTPS):     no
    WebSockets:            yes
    RabbitMQ:              no
    MQTT:                  no
    Unix Sockets:          yes
    Nanomsg:               no
Plugins:
    Echo Test:             yes
    Streaming:             yes
    Video Call:            yes
    SIP Gateway:           no
    NoSIP (RTP Bridge):    yes
    Audio Bridge:          no
    Video Room:            yes
    Voice Mail:            no
    Record&Play:           yes
    Text Room:             yes
    Lua Interpreter:       no
    Duktape Interpreter:   no
Event handlers:
    Sample event handler:  no
    WebSocket ev. handler: yes
    RabbitMQ event handler:no
    MQTT event handler:    no
    Nanomsg event handler: no
    GELF event handler:    yes
External loggers:
    JSON file logger:      no
JavaScript modules:        no
```

### CODE REVIEW

File: Containers/talk/Dockerfile [hunk 1]
```diff
@@ -1,44 +1,45 @@
...
 FROM strukturag/nextcloud-spreed-signaling:1.1.2 as signaling
+
+FROM alpine:3.18.2 as janus
+ARG JANUS_VERSION=v0.13.4
+WORKDIR /src
+RUN apk add --no-cache \
+        ca-certificates \
+        git autoconf automake build-base pkgconfig libtool util-linux \
+        glib-dev zlib-dev openssl-dev jansson-dev libnice-dev libconfig-dev libsrtp-dev gengetopt-dev \
+        libwebsockets-dev; \
```
Comment (Zoey2936) [inline Containers/talk/Dockerfile:11]: we maybe need to change to optional modules

File: Containers/talk/Dockerfile [hunk 2]
```diff
@@ -1,44 +1,45 @@
...
         netcat-openbsd \
-        shadow \
-        util-linux \
-        build-base \
-        wget \
-        lua5.3-dev \
-        luarocks5.3; \
+        \
+        glib zlib libssl3 libcrypto3 jansson libnice libconfig libsrtp \
+        libwebsockets \
```
Comment (Zoey2936) [inline Containers/talk/Dockerfile:37]: we maybe need to change to optional modules

File: Containers/talk/Dockerfile (hunk 1 above)
Comment (Zoey2936) [reply Containers/talk/Dockerfile:11]: the structureag image also contains: `curl-dev libogg-dev`

File: Containers/talk/Dockerfile (hunk 2 above)
Comment (Zoey2936) [reply Containers/talk/Dockerfile:37]: the structureag image also contains: `libcurl libogg`

File: Containers/talk/Dockerfile (hunk 1 above)
Comment (Zoey2936) [reply Containers/talk/Dockerfile:11]: ogg adds `Voice Mail` and curl `Sample event handler`/`Sample event handler`

File: Containers/talk/Dockerfile (hunk 2 above)
Comment (Zoey2936) [reply Containers/talk/Dockerfile:37]: ogg adds `Voice Mail` and curl `Sample event handler`/`Sample event handler`

File: Containers/talk/Dockerfile (hunk 1 above)
Comment (Zoey2936) [reply Containers/talk/Dockerfile:11]: If I read https://github.com/strukturag/nextcloud-spreed-signaling#setup-of-janus, we don't need those three modules

File: Containers/talk/Dockerfile (hunk 2 above)
Comment (Zoey2936) [reply Containers/talk/Dockerfile:37]: If I read https://github.com/strukturag/nextcloud-spreed-signaling#setup-of-janus, we don't need those three modules

**Comment by Zoey2936** [comment]
I've tried building janus 1.x by just changing the ARG and it works the same way, so if talk android at some point supports janus 1.x we just need to edit the talk update workflow

File: Containers/talk/Dockerfile (hunk 1 above)
Comment (szaimen) [reply Containers/talk/Dockerfile:11]: all right, then lets remove them.

File: Containers/talk/Dockerfile (hunk 2 above)
Comment (szaimen) [reply Containers/talk/Dockerfile:37]: all right, then lets remove them.

**Comment by szaimen** [comment]
> I've tried building janus 1.x by just changing the ARG and it works the same way, so if talk android at some point supports janus 1.x we just need to edit the talk update workflow

Cool!

File: Containers/talk/supervisord.conf [hunk 3]
```diff
@@ -28,7 +27,7 @@ stdout_logfile_maxbytes=0
 stderr_logfile=/dev/stderr
 stderr_logfile_maxbytes=0
 # debug-level 3 means warning
-command=janus --config=/etc/janus/janus.jcfg --disable-colors --log-stdout --full-trickle --debug-level 3
+command=janus --config=/usr/local/etc/janus/janus.jcfg --disable-colors --log-stdout --full-trickle --debug-level 3
```
Comment (szaimen) [inline Containers/talk/supervisord.conf:30]: Is --full-trickle supported by janus 0.x?

File: Containers/talk/Dockerfile (hunk 2 above)
Comment (szaimen) [reply Containers/talk/Dockerfile:37]: ah I see, sorry

File: Containers/talk/Dockerfile (hunk 1 above)
Comment (szaimen) [reply Containers/talk/Dockerfile:11]: ah I see, sorry

File: Containers/talk/supervisord.conf (hunk 3 above)
Comment (Zoey2936) [reply Containers/talk/supervisord.conf:30]: no idea

Comment (szaimen) [reply Containers/talk/supervisord.conf:30]: yes, looks like it.

File: Containers/talk/Dockerfile [hunk 4]
```diff
@@ -1,41 +1,42 @@
...
+        \
+        shadow; \
     useradd --system talk; \
-    luarocks-5.3 install luajson; \
-    luarocks-5.3 install ansicolors; \
```
Comment (szaimen) [inline Containers/talk/Dockerfile:30]: why arent they needed anymore?

Comment (Zoey2936) [reply Containers/talk/Dockerfile:30]: we build janus without lua support

**Comment by szaimen** [comment]
@Zoey2936 I tried to rebase and added some small changes. Hope I did not break something?

**Comment by Zoey2936** [comment]
what did you changed?

**Comment by szaimen** [comment]
> what did you changed?

I hope nothing. Can you confirm?

**Comment by szaimen** [comment]
> at the moment not, just build the image and see if the janus build works

all right 👍

File: Containers/talk/Dockerfile (hunk 4 above)
Comment (szaimen) [reply Containers/talk/Dockerfile:30]: that makes sense, cool!

File: Containers/talk/Dockerfile [hunk 5]
```diff
@@ -1,41 +1,62 @@
...
 FROM strukturag/nextcloud-spreed-signaling:1.1.3 as signaling
-FROM coturn/coturn:4.6.2-alpine3.18
-USER root
+FROM alpine:3.18.2 as janus
 
-COPY --from=nats /nats-server /usr/local/bin/nats-server
-COPY --from=signaling /usr/bin/nextcloud-spreed-signaling /usr/local/bin/nextcloud-spreed-signaling
+ARG JANUS_VERSION=v0.13.4
+WORKDIR /src
+RUN apk add --no-cache \
```
Comment (szaimen) [inline Containers/talk/Dockerfile:7]:
```suggestion
RUN set -ex; \
    apk add --no-cache \
```

**Comment by Zoey2936** [comment]
not sure if you know if bit janus 0.x still gets updates

**Comment by szaimen** [comment]
> not sure if you know it but janus 0.x still gets updates

yes, that is what I also found out recently 👍

**Review by szaimen — APPROVED** [review]

**Comment by szaimen** [comment]
Thank you very much for tackling this @Zoey2936 ! 💙

**Comment by Zoey2936** [comment]
can you please show me full logs?

**Comment by Zoey2936** [comment]
other question is that an error? or a warning?

**Comment by szaimen** [comment]
Possibly fixed with https://github.com/nextcloud/all-in-one/commit/2c26a1957c3dece0e4553e614349f61240872fbd

**Comment by Zoey2936** [comment]
I think if you add the -dev libary you also need the normal and the build flags are not needed since mqtt is no added as libary, as I think

**Comment by szaimen** [comment]
Normal libusrsctp is included in -dev afaics
https://pkgs.alpinelinux.org/package/v3.18/community/x86_64/libusrsctp-dev

**Comment by Zoey2936** [comment]
but it is not inside the normal container, only the janus build part

**Comment by szaimen** [comment]
ah true, you are right.

**Comment by szaimen** [comment]
https://github.com/nextcloud/all-in-one/commit/2f842e7266eecbbfc8146c953b1500a5001a357c

**Comment by Zoey2936** [comment]
did you tested it with talk android?

**Comment by szaimen** [comment]
> did you tested it with talk android?

Yes and it worked

**Pull request merged**